src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -203,13 +203,18 @@ private static void EndInvokeDotNetAfterTask(Task task, JSRuntime jsRuntime, in
                 var parameterType = parameterTypes[index];
                 if (reader.TokenType == JsonTokenType.StartObject && IsIncorrectDotNetObjectRefUse(parameterType, reader))
                 {
+                    // We don't strictly have to clear the ByteArraysToBeRevived because it will auto-clear at the start of the next call,
+                    // but since we know we won't be using it for this call, we might as well clear it
+                    jsRuntime.ResetByteArraysToBeRevived();
                     throw new InvalidOperationException($"In call to '{methodIdentifier}', parameter of type '{parameterType.Name}' at index {(index + 1)} must be declared as type 'DotNetObjectRef<{parameterType.Name}>' to receive the incoming value.");
                 }
 
                 suppliedArgs[index] = JsonSerializer.Deserialize(ref reader, parameterType, jsRuntime.JsonSerializerOptions);
                 index++;
             }
 
+            jsRuntime.ResetByteArraysToBeRevived();
```

Comment:
Once we are done de-serializing the arguments, should we check to ensure that we've consumed all incoming byte arrays and throw if that's not the case?

I'm worried that otherwise we are dropping data and I don't see any valid scenario where it would be ok to receive M byte arrays and consume only N of them during deserialization.

Reply:
It is legit to send JSON properties that aren't consumed on the receiving side. Those properties might contain an `ArrayBuffer`. It would be a further change of design to assume the .NET object shape completely covers the JS object shape, rather than allowing any subset.

I know it's *wasteful* to send properties you aren't consuming, but it would be odd to consider it an error based on the type of the property value.

Reply:
Fair enough, that's a good point and a valid scenario. I think this just needs a comment then for our future selves.